tests/unit/cartography/intel/sentinelone/test_application.py

```diff
@@ -0,0 +1,428 @@
+    """
+    result = transform_applications(
+        [
+            {
+                "applicationName": "Test App With Spaces",
+                "applicationVendor": "Test Vendor With Spaces",
+            },
+            {
+                "applicationName": "Test-App/With$Special#Characters",
+                "applicationVendor": "Test@Vendor&With%Special*Characters",
+            },
+        ]
+    )
+
+    # Verify ID normalization
+    assert result[0]["id"] == "test_vendor_with_spaces:test_app_with_spaces"
+    assert (
+        result[1]["id"]
+        == "testvendorwithspecialcharacters:testappwithspecialcharacters"
+    )
+
+
+def test_application_version_id_generation():
+    """
+    Test that application version IDs are generated consistently with proper normalization
+    """
+    result = transform_application_versions(
+        [
+            {
+                "applicationName": "Test App",
+                "applicationVendor": "Test Vendor",
+                "version": "1.2.3-beta",
+                "endpointUuid": AGENT_UUID_1,
+            },
+            {
+                "applicationName": "Test App",
+                "applicationVendor": "Test Vendor",
+                "version": "2.0.0+build.123",
+                "endpointUuid": AGENT_UUID_2,
+            },
+        ]
+    )
+
+    # Verify version ID normalization includes version
+    assert result[0]["id"] == "test_vendor:test_app:1.2.3-beta"
+    assert result[1]["id"] == "test_vendor:test_app:2.0.0build.123"
+
+
+@patch("cartography.intel.sentinelone.application.get_paginated_results")
+def test_get_application_data(mock_get_paginated_results):
+    """
+    Test that get_application_data calls API with correct parameters and returns data
+    """
+    # Mock API response
+    mock_get_paginated_results.return_value = APPLICATIONS_DATA
+
+    # Call function
+    result = get_application_data(TEST_ACCOUNT_ID, TEST_API_URL, TEST_API_TOKEN)
+
+    # Verify API was called with correct parameters
+    mock_get_paginated_results.assert_called_once_with(
+        api_url=TEST_API_URL,
+        endpoint="/web/api/v2.1/application-management/inventory",
+        api_token=TEST_API_TOKEN,
+        params={
+            "accountIds": TEST_ACCOUNT_ID,
+            "limit": 1000,
+        },
+    )
+
+    # Verify return value
+    assert result == APPLICATIONS_DATA
+
+
+@patch("cartography.intel.sentinelone.application.get_paginated_results")
+def test_get_application_data_empty_response(mock_get_paginated_results):
+    """
+    Test that get_application_data handles empty API response
+    """
+    # Mock empty API response
+    mock_get_paginated_results.return_value = []
+
+    # Call function
+    result = get_application_data(TEST_ACCOUNT_ID, TEST_API_URL, TEST_API_TOKEN)
+
+    # Verify return value
+    assert result == []
+
+
+@patch("cartography.intel.sentinelone.application.get_paginated_results")
+def test_get_application_installs(mock_get_paginated_results):
```

Comment:
Nit: I don't think we need this test since the majority of the functionality is patched out, but if you feel it's useful to exercise the code path we can keep it